galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java

```diff
@@ -229,14 +241,13 @@ public ServerProcess getOneTerminatedServer() throws GalvanFailureException {
   public void serverBecameActive(ServerProcess server) {
     synchronized (this.sharedLockState) {
       localAssert(null == this.activeServer, server);
-      boolean didRemove = this.unknownRunningServers.remove(server);
-      boolean wasPassive = false;
-      if (!didRemove) {
-        didRemove = this.passiveServers.remove(server);
-        wasPassive = true;
-      }
-      this.logger.output("serverBecameActive: " + server + " was passive:" + wasPassive);
-      localAssert(didRemove, server);
+      boolean wasUnknown = this.unknownRunningServers.remove(server);
+      boolean wasPassive = this.passiveServers.remove(server);
+      boolean wasDiagnostic = this.diagnosticServers.remove(server);
+      String previousState = wasUnknown ? "unknown" : wasPassive ? "passive" : "diagnostic";
+      this.logger.output("serverBecameActive: " + server + "(was " + previousState + ")");
+
+      localAssert(wasUnknown || wasPassive || wasDiagnostic, server);
```

Comment:
can servers go from diagnostic to active without restart?  If not, I would reconsider this logic.

Reply:
They cannot. I'll revert this part.